// @libs/common/dist/timers/stopwatch.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class Stopwatch {
    get elapsedMilliseconds() {
        return this.convertToMilliseconds(this.elapsed);
    }
    get elapsed() {
        return this._isRunning ? process.hrtime(this._start) : this._current;
    }
    reset() {
        this._start = null;
        this._current = null;
        this._isRunning = false;
        return this;
    }
    restart() {
        this.reset();
        this.start();
        return this;
    }
    start() {
        if (!this._start) {
            this._start = process.hrtime();
        }
        this._isRunning = true;
        return this;
    }
    pause() {
        this._current = process.hrtime(this._start);
        this._isRunning = false;
        return this;
    }
    convertToMilliseconds(hrtime) {
        const nanoseconds = (hrtime[0] * 1e9) + hrtime[1];
        const milliseconds = nanoseconds / 1e6;
        return milliseconds;
    }
}
exports.Stopwatch = Stopwatch;
